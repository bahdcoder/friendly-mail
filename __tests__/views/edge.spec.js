const EdgeRenderEngine = require('../../src/Views/Edge')

describe('The EdgeRenderEngine', () => {
  it('it instantiates properly', () => {
    const config = {
      viewEngine: 'edge'
    }

    expect(new EdgeRenderEngine(config)).toMatchSnapshot()
  })

  it('can render the emails content for all three email types', () => {
    const config = {
      viewEngine: 'edge',
      views: '__tests__/__mocks__/mails'
    }

    expect(
      new EdgeRenderEngine(config).render('reset-password', {
        username: 'bahdcoder'
      })
    ).toMatchSnapshot()
  })
})
