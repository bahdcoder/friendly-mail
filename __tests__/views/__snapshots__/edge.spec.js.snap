// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EdgeRenderEngine can render the emails content for all three email types 1`] = `
Object {
  "html": "<strong>
  HEY bahdcoder
</strong>
<br>
RESET YOUR PASSWORD
",
  "text": "hey bahdcoder, please RESET YOUR PASSWORD.
",
  "watchHtml": "hey bahdcoder, please RESET YOUR PASSWORD.
",
}
`;

exports[`The EdgeRenderEngine it instantiates properly 1`] = `
EdgeRenderEngine {
  "Config": Object {
    "viewEngine": "edge",
  },
  "enginesExtensionsMap": Object {
    "edge": "edge",
    "handlebars": "hbs",
  },
}
`;
